network/endpoint_linux.go

```diff
@@ -278,3 +284,120 @@ func deleteRoutes(interfaceName string, routes []RouteInfo) error {
+	if existingEpFromRepository == nil {
+		log.Printf("[updateEndpointImpl] Endpoint cannot be updated as it does not exist.")
+		err = errEndpointNotFound
+		return nil, err
+	}
+
+	netns := existingEpFromRepository.NetworkNameSpace
+	// Network namespace for the container interface has to be specified
+	if netns != "" {
+		// Open the network namespace.
+		log.Printf("[updateEndpointImpl] Opening netns %v.", netns)
+		ns, err = OpenNamespace(netns)
+		if err != nil {
+			return nil, err
+		}
+		defer ns.Close()
+
+		// Enter the container network namespace.
+		log.Printf("[updateEndpointImpl] Entering netns %v.", netns)
+		if err = ns.Enter(); err != nil {
+			return nil, err
+		}
+
+		// Return to host network namespace.
+		defer func() {
+			log.Printf("[updateEndpointImpl] Exiting netns %v.", netns)
+			if err := ns.Exit(); err != nil {
+				log.Printf("[updateEndpointImpl] Failed to exit netns, err:%v.", err)
+			}
+		}()
+	} else {
+		log.Printf("[updateEndpointImpl] Endpoint cannot be updated as the network namespace does not exist: Epid: %v", existingEpInfo.Id)
+		err = errNamespaceNotFound
+		return nil, err
+	}
+
+	log.Printf("[updateEndpointImpl] Going to update routes in netns %v.", netns)
+	if err = updateRoutes(existingEpInfo, targetEpInfo); err != nil {
+		return nil, err
+	}
+
+	// Create the endpoint object.
+	ep = &endpoint{
+		Id: existingEpInfo.Id,
+	}
+
+	// Update existing endpoint state with the new routes to persist
+	for _, route := range targetEpInfo.Routes {
+		ep.Routes = append(ep.Routes, route)
+	}
+
+	return ep, nil
+}
+
+func updateRoutes(existingEp *EndpointInfo, targetEp *EndpointInfo) error {
+	log.Printf("Updating routes for the endpoint %+v.", existingEp)
+	log.Printf("Target endpoint is %+v", targetEp)
+
+	existingRoutes := make(map[string]RouteInfo)
+	targetRoutes := make(map[string]RouteInfo)
+	var tobeDeletedRoutes []RouteInfo
+	var tobeAddedRoutes []RouteInfo
+
+	// we should not remove default route from container if it exists
+	// we do not support enable/disable snat for now
+	defaultDst := net.ParseIP("0.0.0.0")
+
+	for _, route := range existingEp.Routes {
+		if route.Dst.IP.String() != defaultDst.String() {
+			existingRoutes[route.Dst.String()] = route
+		}
+	}
+
+	for _, route := range targetEp.Routes {
+		targetRoutes[route.Dst.String()] = route
+	}
+
+	for _, existingRoute := range existingRoutes {
+		dst := existingRoute.Dst.String()
+		if _, ok := targetRoutes[dst]; !ok {
+			tobeDeletedRoutes = append(tobeDeletedRoutes, existingRoute)
+			log.Printf("Adding following route to the tobeDeleted list: %+v", existingRoute)
+		}
+	}
+
+	for _, targetRoute := range targetRoutes {
+		dst := targetRoute.Dst.String()
+		if _, ok := existingRoutes[dst]; !ok {
+			tobeAddedRoutes = append(tobeAddedRoutes, targetRoute)
+			log.Printf("Adding following route to the tobeAdded list: %+v", targetRoute)
+		}
+
+	}
+
+	err := deleteRoutes(existingEp.IfName, tobeDeletedRoutes)
```

Comment:
it should be 
if err := deleteRoutes(existingEp.IfName, tobeDeletedRoutes); err != nil {

}